drivers/Watchdog.h

```diff
@@ -44,62 +47,74 @@ namespace mbed {
+     * Assert for multiple calls of start
      */
-    watchdog_status_t start(const uint32_t timeout);
-
-
-    /** Refreshes the watchdog timer.
-     *
-     * This function should be called periodically before the watchdog times out.
-     * Otherwise, the system is reset.
-     *
-     * If the watchdog timer is not currently running this function does nothing
-     */
-    void kick();
-
+    void start();
 
     /** Stops the watchdog timer
      *
      * Calling this function will attempt to disable any currently running
      * watchdog timers if supported by the current platform.
      *
-     * @return Returns WATCHDOG_STATUS_OK if the watchdog timer was succesfully
-     *         stopped, or if the timer was never started. Returns
-     *         WATCHDOG_STATUS_NOT_SUPPORTED if the watchdog cannot be disabled
-     *         on the current platform.
+     * Assert with out called start
      */
-    watchdog_status_t stop();
+    void stop();
 
-
-    /** Get the watchdog timer refresh value
+    /** Refreshes the watchdog timer.
      *
-     * This function returns the refresh timeout of the watchdog timer.
+     * This function should be called periodically before the watchdog times out.
+     * Otherwise, the system is reset.
      *
-     * @return Reload value for the watchdog timer in milliseconds.
+     * If the watchdog timer is not currently running this function does nothing
      */
-    uint32_t reload_value() const;
-
+    void kick();
 
-    /** Get the maximum refresh value for the current platform in milliseconds
+    /** Interface API to mbed_watchdog_manager to monitor user/threads.
+     *
+     * This function should be called from mbed_watchdog_manager_kick to monitor all the
+     * user/threads alive state.
      *
-     * @return Maximum refresh value supported by the watchdog for the current
-     *         platform in milliseconds
+     * Otherwise, the system is reset.
      */
-    static uint32_t max_timeout();
+    void is_alive();
+protected :
+
+    /** add_to_list is used to store the registered user into List.
+      * This API is only used to call from start.
+      *
+      *  @return status ture if it stored the address of class instance
+      *                 successfully. false if one of the input
+      *                 parameters is already have entry in the list
+      */
+    bool add_to_list();
+
+    /** Remove from list is uesed to remove the entry from the list.
+      * This API is only used to call from stop.
+      *
+      *  @return status true if register to watchdog timer was succeeded
+      *                 successfully. false if tid is not registered
+      *                 and try to do unregister
+      */
+    bool  remove_from_list();
+private:
+    uint32_t _max_timeout; //_max_timeout initialized via constructor while creating instance of this class
+    std::string str;//To store the details of user
```

Comment:
Using the string class here will use dynamic memory and pull in extra code. I recommend using a static C string instead as the Thread class does.